src/ripple/app/main/Main.cpp

```diff
@@ -517,33 +517,34 @@ run(int argc, char** argv)
         }
 
         using namespace boost::filesystem;
-        DatabaseCon::Setup dbSetup = setup_DatabaseCon(*config);
+        DatabaseCon::Setup const dbSetup = setup_DatabaseCon(*config);
         path dbPath = dbSetup.dataDir / TxDBName;
 
         try
         {
             uintmax_t const dbSize = file_size(dbPath);
             assert(dbSize != static_cast<uintmax_t>(-1));
 
+            if (auto available = space(dbPath.parent_path()).available;
+                available < dbSize)
             {
-                auto available = space(dbPath.parent_path()).available;
-                if (available < dbSize)
-                {
-                    std::cerr
-                        << "The database filesystem must have at least as "
-                           "much free space as the size of "
-                        << dbPath.string() << ", which is " << dbSize
-                        << " bytes. Only " << available
-                        << " bytes are available.\n";
-                    return -1;
-                }
+                std::cerr << "The database filesystem must have at least as "
+                             "much free space as the size of "
+                          << dbPath.string() << ", which is " << dbSize
+                          << " bytes. Only " << available
+                          << " bytes are available.\n";
+                return -1;
             }
 
             auto txnDB = std::make_unique<DatabaseCon>(
                 dbSetup, TxDBName, TxDBPragma, TxDBInit);
             auto& session = txnDB->getSession();
             std::uint32_t pageSize;
 
+            // Only the most trivial databases will fit in memory on typical
+            // (recommended) software. Force temp files to be written to disk
+            // regardless of the config settings.
+            session << boost::format(CommonDBPragmaTemp) % "file";
```

Comment:
The original behavior needs to come back--namely, that the actual temp directory be specified. Otherwise, SQLIte will by default use the directory in which the database is.

Alternately, I think that getting rid of this feature altogether makes a lot of sense. If somebody *really* wants to VACUUM a database, it's easier to use the command line tool.